### PR TITLE
WIP: serial communication, added Projects and implementation of some functions

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
mainly to Datareader and datawriter
replaces #177

### CODE REVIEW

File: Windows.Devices.SerialCommunication/Nuget.Windows.Devices.SerialCommunication.DELIVERABLES/Nuget.Windows.Devices.SerialCommunication.DELIVERABLES.nuproj [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    <LicenseUrl>
+    </LicenseUrl>
+    <Copyright>Copyright (c) 2017 The nanoFramework project contributors</Copyright>
+    <Tags>
+    </Tags>
+    <IconUrl>https://secure.gravatar.com/avatar/97d0e092247f0716db6d4b47b7d1d1ad</IconUrl>
+    <GenerateSymbolPackage>false</GenerateSymbolPackage>
+  </PropertyGroup>
+  <Import Project="$(NuProjPath)\NuProj.targets" />
+</Project>
```
Comment (josesimoes) [inline Windows.Devices.SerialCommunication/Nuget.Windows.Devices.SerialCommunication.DELIVERABLES/Nuget.Windows.Devices.SerialCommunication.DELIVERABLES.nuproj]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Devices.SerialCommunication/Nuget.Windows.Devices.SerialCommunication/Nuget.Windows.Devices.SerialCommunication.nuproj [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    <ProjectUrl>https://github.com/nanoframework/nf-class-libraries</ProjectUrl>
+    <LicenseUrl>
+    </LicenseUrl>
+    <Copyright>Copyright (c) 2017 The nanoFramework project contributors</Copyright>
+    <Tags>nanoFramework C# csharp netmf netnf Windows.Devices.SerialCommunication</Tags>
+    <IconUrl>https://secure.gravatar.com/avatar/97d0e092247f0716db6d4b47b7d1d1ad</IconUrl>
+    <GenerateSymbolPackage>false</GenerateSymbolPackage>
+  </PropertyGroup>
+  <Import Project="$(NuProjPath)\NuProj.targets" />
+</Project>
```
Comment (josesimoes) [inline Windows.Devices.SerialCommunication/Nuget.Windows.Devices.SerialCommunication/Nuget.Windows.Devices.SerialCommunication.nuproj]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Devices.SerialCommunication/Windows.Devices.SerialCommunication.nfproj [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      <SpecificVersion>True</SpecificVersion>
+    </Reference>
+  </ItemGroup>
+  <Import Project="$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets" Condition="Exists('$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets')" />
+  <ProjectExtensions>
+    <ProjectCapabilities>
+      <ProjectConfigurationsDeclaredAsItems />
+    </ProjectCapabilities>
+  </ProjectExtensions>
+</Project>
```
Comment (josesimoes) [inline Windows.Devices.SerialCommunication/Windows.Devices.SerialCommunication.nfproj]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Devices.SerialCommunication/packages.config [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="nanoFramework.CoreLibrary" version="1.0.0-preview028" targetFramework="net46" />
+  <package id="nanoFramework.Runtime.Events" version="1.0.0-preview017" targetFramework="net46" />
+  <package id="NuProj" version="0.20.4-beta" developmentDependency="true" />
+  <package id="NuProj.Common" version="0.20.4-beta" developmentDependency="true" />
+</packages>
```
Comment (josesimoes) [inline Windows.Devices.SerialCommunication/packages.config]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Storage.Streams/Nuget.Windows.Storage.Streams.DELIVERABLES/Nuget.Windows.Storage.Streams.DELIVERABLES.nuproj [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    <LicenseUrl>
+    </LicenseUrl>
+    <Copyright>Copyright (c) 2017 The nanoFramework project contributors</Copyright>
+    <Tags>
+    </Tags>
+    <IconUrl>https://secure.gravatar.com/avatar/97d0e092247f0716db6d4b47b7d1d1ad</IconUrl>
+    <GenerateSymbolPackage>false</GenerateSymbolPackage>
+  </PropertyGroup>
+  <Import Project="$(NuProjPath)\NuProj.targets" />
+</Project>
```
Comment (josesimoes) [inline Windows.Storage.Streams/Nuget.Windows.Storage.Streams.DELIVERABLES/Nuget.Windows.Storage.Streams.DELIVERABLES.nuproj]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Storage.Streams/Nuget.Windows.Storage.Streams/Nuget.Windows.Storage.Streams.nuproj [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    <ProjectUrl>https://github.com/nanoframework/nf-class-libraries</ProjectUrl>
+    <LicenseUrl>
+    </LicenseUrl>
+    <Copyright>Copyright (c) 2017 The nanoFramework project contributors</Copyright>
+    <Tags>nanoFramework C# csharp netmf netnf Windows.Storage.Streams</Tags>
+    <IconUrl>https://secure.gravatar.com/avatar/97d0e092247f0716db6d4b47b7d1d1ad</IconUrl>
+    <GenerateSymbolPackage>false</GenerateSymbolPackage>
+  </PropertyGroup>
+  <Import Project="$(NuProjPath)\NuProj.targets" />
+</Project>
```
Comment (josesimoes) [inline Windows.Storage.Streams/Nuget.Windows.Storage.Streams/Nuget.Windows.Storage.Streams.nuproj]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Storage.Streams/Windows.Storage.Streams.nfproj [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      <SpecificVersion>True</SpecificVersion>
+    </Reference>
+  </ItemGroup>
+  <Import Project="$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets" Condition="Exists('$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets')" />
+  <ProjectExtensions>
+    <ProjectCapabilities>
+      <ProjectConfigurationsDeclaredAsItems />
+    </ProjectCapabilities>
+  </ProjectExtensions>
+</Project>
```
Comment (josesimoes) [inline Windows.Storage.Streams/Windows.Storage.Streams.nfproj]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Storage.Streams/packages.config [hunk 8]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="nanoFramework.CoreLibrary" version="1.0.0-preview028" targetFramework="net46" />
+  <package id="nanoFramework.Runtime.Events" version="1.0.0-preview017" targetFramework="net46" />
+  <package id="NuProj" version="0.20.4-beta" developmentDependency="true" />
+  <package id="NuProj.Common" version="0.20.4-beta" developmentDependency="true" />
+</packages>
```
Comment (josesimoes) [inline Windows.Storage.Streams/packages.config]: please add an empty line at the end of the file (Git needs this for better comparing files)

File: Windows.Storage.Streams/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.27")]
```
Comment (josesimoes) [inline Windows.Storage.Streams/Properties/AssemblyInfo.cs]: Need to keep these in sync with the NuGet version, so these would be 1.0.0.1

Comment (josesimoes) [reply Windows.Storage.Streams/Properties/AssemblyInfo.cs]: (BTW, is the version of the SerialComm assembly in sync with the NuGet version? please check that too 😉 )

Comment (josesimoes) [reply Windows.Storage.Streams/Properties/AssemblyInfo.cs]: It's still showing 1.0.0.27... Needs to be 1.0.0.1...

File: Windows.Storage.Streams/packages.config (hunk 8 above)
Comment (networkfusion) [reply Windows.Storage.Streams/packages.config]: The lines were added... are they not showing?!

Comment (josesimoes) [reply Windows.Storage.Streams/packages.config]: Yes. The empty lines are there now. 😃 
What I was requesting in the other comment wad to change the assembly version and file attribute to 1
0.0.1 to be on sync with the NuGet package id.

**Pull request merged**